### PR TITLE
[PD-2087] Add UI Form Documentation

### PR DESCRIPTION
**Pull request by fezick**
Adds documentation for form usage in the UI

### CODE REVIEW

File: doc/source/developers/styleguides/forms.rst [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+**Form Validation**:
+
+* Validation errors should be able to stack when there are multiple errors
+* Non field form errors should appear at the top of a form in a red alert
+* All Validation warnings should preserve any input already entered in a form
+* Validation warnings should replace any help text on the form
+* Validation details (if applicable) should be hidden by default
+ 
+Alignement
```
Comment (aspidites) [inline doc/source/developers/styleguides/forms.rst]: Alignement -> Alignment

File: doc/source/developers/styleguides/forms.rst [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+Field Validation
+================
+
+Changes made to a field in error:
+
+* input border turns red
+* label turns red
+* help text is hidden (if applicable)
+* a warning icon and error description are shown immediatly below the fold. If
```
Comment (aspidites) [inline doc/source/developers/styleguides/forms.rst]: immediatly -> immediately

File: doc/source/developers/styleguides/forms.rst [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  there are multiple errors, each one gets its own line below the field
+* if additional details are available for an error, they should be placed in a 
+  hidden div below the error.
+  
+  * To access this div, a tree arrow whould be placed aligned right on the
```
Comment (aspidites) [inline doc/source/developers/styleguides/forms.rst]: whould -> would

File: doc/source/developers/styleguides/forms.rst (hunk 1 above)
Comment (fezick) [inline doc/source/developers/styleguides/forms.rst]: I like my creative splleing

Comment (tdphillips) [inline doc/source/developers/styleguides/forms.rst]: It's in Ye Olde English, can't you tell?

**Pull request merged**